Tools/CMake/AMReXParallelBackends.cmake

```
@@ -192,19 +194,27 @@ if (AMReX_HIP)
 
    set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
 
-   find_package(HIP REQUIRED)
+   find_package(hip REQUIRED)
 
-   if("${HIP_COMPILER}" STREQUAL "hcc")
-      message(FATAL_ERROR "Using (deprecated) HCC compiler: please update ROCm")
-   endif()
-
-   if(HIP_FOUND)
-      message(STATUS "Found HIP: ${HIP_VERSION}")
-      message(STATUS "HIP: Platform=${HIP_PLATFORM} Compiler=${HIP_COMPILER}")
+   if(hip_FOUND)
+      message(STATUS "Found HIP: ${hip_VERSION}")
+      message(STATUS "HIP: Platform=${hip_PLATFORM} Compiler=${hip_COMPILER}")
    else()
       message(FATAL_ERROR "Could not find HIP."
          " Ensure that HIP is either installed in /opt/rocm/hip or the variable HIP_PATH is set to point to the right location.")
    endif()
+   
+   # avoid forcing the rocm LLVM flags on a gfortran
+   # https://github.com/ROCm-Developer-Tools/HIP/issues/2275
+   if(AMReX_FORTRAN)
+       message(WARNING "As of rocM/HIP <= 4.2.0, Fortran support might be flaky.\n"
+                       "Especially, we cannot yet support reloctable device code (RDC)."
+                       "See https://github.com/ROCm-Developer-Tools/HIP/issues/2275 "
+                       "and https://github.com/AMReX-Codes/amrex/pull/2031 "
+                       "for details.")
+   else()
+       target_link_libraries(amrex PUBLIC hip::device)
```

`clang++ -x hip` defaults to C++11: https://github.com/ROCm-Developer-Tools/HIP/issues/2278

Need to work-around by either adding to the `COMPILE_OPTIONS` property or fixing `CXX_STANDARD` to C++14 (somewhat different to the "minimum C++14" that we do today).
```suggestion
       target_link_libraries(amrex PUBLIC hip::device)
       
       # work-around for https://github.com/ROCm-Developer-Tools/HIP/issues/2278
       # the CXX_STANDARD property always adds the flag, even if the compiler default fulfills it
       set_property(TARGET amrex PROPERTY CXX_STANDARD 14)
```